app/src/androidTest/java/com/example/habitsmasher/MainActivityTest.java

```diff
@@ -1409,6 +1603,41 @@ private void setDaysInAddHabitDialogBox(){
         _solo.clickOnView(_solo.getView(R.id.friday_button));
     }
 
+    private void setCurrentDayInAddHabitDialogBox(){
+        String currentDay = getCurrentDay();
+
+        if (currentDay.equals("MO")){
+            _solo.clickOnView(_solo.getView(R.id.monday_button));
+        }
+        else if (currentDay.equals("TU")){
+            _solo.clickOnView(_solo.getView(R.id.tuesday_button));
+        }
+        else if (currentDay.equals("WE")){
+            _solo.clickOnView(_solo.getView(R.id.wednesday_button));
+        }
+        else if (currentDay.equals("TU")){
+            _solo.clickOnView(_solo.getView(R.id.thursday_button));
+        }
+        else if (currentDay.equals("FR")){
+            _solo.clickOnView(_solo.getView(R.id.friday_button));
+        }
+        else if (currentDay.equals("SA")){
+            _solo.clickOnView(_solo.getView(R.id.saturday_button));
+        }
+        else if (currentDay.equals("SU")){
+            _solo.clickOnView(_solo.getView(R.id.sunday_button));
+        }
+    }
+
+    private void setNonCurrentDayInAddHabitDialogBox(){
+        if (getCurrentDay().equals("WE")){
+            _solo.clickOnView(_solo.getView(R.id.thursday_button));
+        }
+        else{
+            _solo.clickOnView(_solo.getView(R.id.wednesday_button));
+        }
+    }
+
```

Comment:
Helpers for setting the current day. Important since main implementation checks this.